### PR TITLE
feat: Add fit-height property to charts

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

The new `fitHeight` property on all charts allows the content area to grow or shrink to fit the outer container. The `height` (mixed charts, area chart) and `size` (pie chart) properties control the minimal allowed height of the content in that case.

### How has this been tested?

New screenshot tests.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/mixed-line-bar-chart/chart-container.tsx [hunk 1]
```diff
@@ -118,6 +121,14 @@ export default function ChartContainer<T extends ChartDataTypes>({
   const containerRef = useMergeRefs(containerMeasureRef, containerRefObject);
   const popoverRef = useRef<HTMLElement | null>(null);
 
+  const plotMeasureRef = useRef<SVGLineElement>(null);
+  const [measuredHeight, setHeight] = useState(0);
+  useResizeObserver(
+    () => plotMeasureRef.current,
+    entry => setHeight(entry.borderBoxHeight)
```
Comment (pan-kot) [inline src/mixed-line-bar-chart/chart-container.tsx:128]: Skip state update when not fitHeight

File: src/pie-chart/utils.ts [hunk 2]
```diff
@@ -55,6 +57,34 @@ export const refreshDimensionsBySize: Record<NonNullable<PieChartProps['size']>,
   },
 };
 
+export function getDimensionsBySize({
```
Comment (timogasda) [inline src/pie-chart/utils.ts:60]: Can you add some comments in the function body to explain a little bit whats happening here and what these numbers mean?

File: src/pie-chart/styles.scss [hunk 3]
```diff
@@ -27,13 +27,30 @@
...
+    height: 100%;
+  }
+}
+
+.chart-container-chart-plot {
+  display: contents;
+
+  &--fit-height {
+    position: absolute;
+    inset: 0;
```
Comment (timogasda) [inline src/pie-chart/styles.scss:52]: This is not supported in Safari 14.0 :( https://caniuse.com/mdn-css_properties_inset

File: src/internal/components/chart-plot/index.tsx [hunk 4]
```diff
@@ -172,9 +172,9 @@ function ChartPlot(
         aria-hidden="false"
         {...plotAria}
         ref={svgRef}
-        width={width}
-        height={height}
```
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/internal/components/chart-plot/index.tsx:176]: Good point. Let's test this one properly. If the explicit numbers are needed we can measure the outer container width/height and provide the explicit values.

**Review by timogasda — APPROVED** [review]

**Review by timogasda — APPROVED** [review]

**Pull request merged**